src/flint/types/fmpz.pyx

```diff
@@ -21,7 +21,7 @@ cdef fmpz_get_intlong(fmpz_t x):
     cdef char * s
     if COEFF_IS_MPZ(x[0]):
         s = fmpz_get_str(NULL, 16, x)
-        v = int(str_from_chars(s), 16)
+        v = int(bytes(s).decode('ascii'), 16)
         libc.stdlib.free(s)
         return v
```

Comment:
I'm not sure why this change seems to have fixed the problem. The function is defined as:
```
   cdef inline str_from_chars(s):
         return bytes(s).decode('ascii')
```
However actually inlining the function here prevents the crash that is only seen on Windows when building with windows-2022 or windows-2025 but not if built on windows-2019... (The crash happens when running on any Windows version but not if the binaries were built on windows-2019.)

Reply:
No, I was tricked. This change does not fix the problem. I fell for the classic printing things makes the segfault go away confusion.

Reply:
Maybe this should use `flint_free`?

Reply:
All the observed problems were fixed by using flint_free when using `fmpz_get_str`. I'm not sure why this issue appears now and not before.

It is probably worth reviewing all other uses of `free` to see if they should be `flint_free`...